src/libexpr/flake/flakeref.cc

```diff
@@ -98,7 +98,7 @@ std::pair<FlakeRef, std::string> parsePathFlakeRefWithFragment(
             to 'baseDir'). If so, search upward to the root of the
             repo (i.e. the directory containing .git). */
 
-        path = absPath(path, baseDir);
+        path = readLink( absPath(path, baseDir) );
```

Comment:
Maybe this should be
```suggestion
        path = canonPath(absPath(path, baseDir), true);
```

but I'm not sure.
I know some of the code, but symlinks freak me out.

This definitely needs some test cases,
- symlink points to flake
- symlink points to directory with a subdir that's a flake (ie the symlink isn't the final component but in between)
- symlink points to a directory in a repo
- symlink points from a repo to another directory. Needs `--impure`? Also test the error.